cfpb/consumerfinance.gov
# TCCP: Add 3-tier card rating based on purchase APR

With this change, cards in the TCCP dataset are given a rating indicating how their purchase APR compares against all other card purchase APRs within the same tier. On the frontend this is rendered as a 3-tier system: pay less interest, pay average intereset, pay more interest.

The actual calculation used to place cards into tiers is based on the percentile of the tier-specific purchase APR:

- if the card APR < 25th percentile of all tier-specific purchase APRs, we label it as "pay less interest"
- if the card APR < 50th percentile of all tier-specific purchase APRs, we label it as "pay average interest"
- otherwise, we label it as "pay more interest"

This change adds a key to the card listing page to help users understand the three tiers.

On the backend this works by first computing summary statistics across all cards (for each tier), including the 25th and 50th percentiles. It leverages the [tailslide](https://github.com/ankane/tailslide) Python package for this, which under the hood uses the PostgreSQL [`PERCENTILE_CONT`](https://www.postgresql.org/docs/14/functions-aggregate.html) function.

## How to test this PR

To test, run a local server and visit:

http://localhost:8000/consumer-tools/credit-cards/explore-cards/cards/

Alternatively, visit:

http://localhost:8000/consumer-tools/credit-cards/explore-cards/cards/?format=json

to see the same data as JSON; note the "stats_all" key which contains the summary stats across the whole dataset, including the percentile cutoffs.

Alternatively, this branch is now live on internal DEV4 for those with access.

## Notes and todos

@contolini, this could use some frontend love on desktop, where it'd be nice if we could make the line break after the APR but before the icon somehow:

|Desktop|Mobile|
|-|-|
|<img width="317" alt="image" src="https://github.com/cfpb/consumerfinance.gov/assets/654645/bd625ee8-a253-4612-8d52-302d5883d843">|<img width="350" alt="image" src="https://github.com/cfpb/consumerfinance.gov/assets/654645/2a21d117-2c40-41b0-ac5c-20f538a59e42">|

Any general frontend suggestions for how I implemented the classnames and CSS are also welcome!

## Checklist
